Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/VehicleRegistrationCertificate.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE ApplicativeDo #-}
+
+module SharedLogic.VehicleRegistrationCertificate
+  ( getAllLinkedRCs,
+    LinkedRCStatus (..),
+  )
+where
+
+import qualified Data.HashMap.Strict as HM
+import qualified Domain.Types.Person as Person
+import Environment
+import Kernel.External.Encryption
+import Kernel.Prelude
+import Kernel.Types.Id
+import Storage.Queries.DriverOnboarding.DriverRCAssociation (buildRcHM)
+import qualified Storage.Queries.DriverOnboarding.DriverRCAssociation as DAQuery
+import qualified Storage.Queries.DriverOnboarding.VehicleRegistrationCertificate as RCQuery
+
+data LinkedRCStatus = LinkedRCStatus
+  { rcNumber :: Text,
+    rcActive :: Bool
+  }
+  deriving (Show, Eq, Read, Generic, ToSchema, ToJSON, FromJSON)
+
+getAllLinkedRCs :: Id Person.Person -> Flow [LinkedRCStatus]
+getAllLinkedRCs driverId = do
+  allLinkedRCs <- DAQuery.findAllLinkedByDriverId driverId
+  rcs <- RCQuery.findAllById (map (.rcId) allLinkedRCs)
+  let activeRcs = buildRcHM allLinkedRCs
+  mapM (getCombinedRcData activeRcs) rcs
+  where
+    getCombinedRcData activeRcs rc = do
+      rcNo <- decrypt rc.certificateNumber
+      return $
+        LinkedRCStatus
+          { rcNumber = rcNo,
+            rcActive = Just True == HM.lookup rc.id.getId activeRcs <&> (.isRcActive)
```

Comment:
your branch not compiled, should be:

`Just True == (HM.lookup rc.id.getId activeRcs <&> (.isRcActive))`

Reply:
Done